manifests/repo/apt.pp

```diff
@@ -1,19 +1,24 @@
 # PRIVATE CLASS: do not use directly
-class mongodb::repo::apt inherits mongodb::repo {
+class mongodb::repo::apt(
+  $location = $mongodb::repo::location,
+  $release  = $mongodb::repo::release,
+  $repos    = $mongodb::repo::repos,
+  $key      = {
+    'id'     => $mongodb::repo::key,
+    'server' => $mongodb::repo::key_server,
+  }
+) inherits mongodb::repo {
   # we try to follow/reproduce the instruction
   # from http://docs.mongodb.org/manual/tutorial/install-mongodb-on-ubuntu/
 
   include ::apt
 
   if($mongodb::repo::ensure == 'present' or $mongodb::repo::ensure == true) {
```

Comment:
Perhaps $ensure should also be a parameter so we could even lose the inheritance and fully decouple.